### PR TITLE
WIP: Complete IS-IS pkt handling

### PR DESCRIPTION
**Pull request by taktv6**

### CODE REVIEW

File: protocols/isis/packet/csnp.go [hunk 1]
```diff
@@ -97,7 +101,7 @@ func NewCSNPs(sourceID types.SourceID, lspEntries []*LSPEntry, maxPDULen int) []
 func newCSNP(sourceID types.SourceID, startLSPID LSPID, endLSPID LSPID, tlvs []TLV) *CSNP {
 	tlvsLen := uint16(0)
 	for i := range tlvs {
-		tlvsLen += uint16(tlvs[i].Length())
+		tlvsLen += uint16(tlvs[i].Length()) + 2
```
Comment (czerwonk) [inline protocols/isis/packet/csnp.go:104]: A constant with an explainatory name would improve the readabiltiy here (magic figure)

**Review by czerwonk — COMMENTED** [review]
LGTM overall (see comment)

**Review by hikhvar — DISMISSED** [review]

File: protocols/isis/packet/csnp.go (hunk 1 above)
Comment (mellowdrifter) [reply protocols/isis/packet/csnp.go:104]: Agreed. Or a comment at least

**Review by mellowdrifter — DISMISSED** [review]

File: protocols/isis/packet/csnp.go (hunk 1 above)
Comment (taktv6) [reply protocols/isis/packet/csnp.go]: Fixed.

**Review by mellowdrifter — APPROVED** [review]

**Review by czerwonk — APPROVED** [review]
LGTM

**Pull request merged**